deno/mod.ts

```diff
@@ -18,20 +18,24 @@ export default class TypedStorage<T> implements ITypedStorage<T> {
 
   constructor(options: TypedStorageOptions = { storage: 'localStorage' }) {
     this.storage = typeof window !== 'undefined' ? window[options.storage] : global[options.storage];
+
+    if (!this.storage) {
+      throw Error('Web Storage API not found.');
+    }
   }
 
   public length(): number {
-    return this.storage.length;
+    return this.storage?.length;
   }
 
   public key<U extends keyof T>(index: number): U {
-    return this.storage.key(index) as U;
+    return this.storage?.key(index) as U;
   }
 
   public getItem<U extends keyof T>(key: U, retrievalMode: RetrievalMode = 'fail'): T[U] | null {
-    const item = this.storage.getItem(key.toString());
+    const item = this.storage?.getItem(key.toString());
 
-    if (item === null) {
+    if (item === null || item === undefined) {
```

Comment:
```suggestion
    if (item == null) {
```

It's a trick I learned from the preact developer to write smaller code.